From 98719ea238213291f63ba54ad4e7bad32adaa208 Mon Sep 17 00:00:00 2001
From: Alex Deucher <alexdeucher@gmail.com>
Date: Thu, 12 Nov 2009 14:01:36 -0500
Patch-mainline: 2.6.33
References: fate#310916
Git-commit: fe6890c3e8019cf1cebce60a86c19180359d3292
Subject: [PATCH 0090/2588] drm/radeon/kms: fix typo in legacy internal tmds
 mode fixup

Call to set active device was missing.

Signed-off-by: Alex Deucher <alexdeucher@gmail.com>
Signed-off-by: Dave Airlie <airlied@redhat.com>
(cherry picked from commit fe6890c3e8019cf1cebce60a86c19180359d3292)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/radeon/radeon_legacy_encoders.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/gpu/drm/radeon/radeon_legacy_encoders.c b/drivers/gpu/drm/radeon/radeon_legacy_encoders.c
index be1dd28..c1de2ed 100644
--- a/drivers/gpu/drm/radeon/radeon_legacy_encoders.c
+++ b/drivers/gpu/drm/radeon/radeon_legacy_encoders.c
@@ -428,7 +428,8 @@ static bool radeon_legacy_tmds_int_mode_fixup(struct drm_encoder *encoder,
 					      struct drm_display_mode *mode,
 					      struct drm_display_mode *adjusted_mode)
 {
-
+	/* set the active encoder to connector routing */
+	radeon_encoder_set_active_device(encoder);
 	drm_mode_set_crtcinfo(adjusted_mode, 0);
 
 	return true;
-- 
1.7.6

